File: WolfDen/wwwroot/ts/Interfaces/ICombatBehaviour.ts
namespace TurnBasedBoardGame {
    export interface ICombatBehaviour extends IExecuteMove {
        Unit: Unit;
        UnitType: UnitType;
        IsValidTarget(unit: Unit): boolean;
        AttackUnit(unit: Unit): void;
        DefendAndCounter(damageSource: ICombatBehaviour, attackMultiplier: number);
        Defend(damageSource: ICombatBehaviour, attackMultiplier: number);
        GetAttackCommand(from: Tile, target: Unit): any;
        CurrentHealth: number;
        MaxHealth: number;
        GetDefenceMultiplier(): number;
        GetPreview(defender: ICombatBehaviour): Preview;
        DefendPreview(attacker: ICombatBehaviour, attackPreview: number): number;
        SetCurrentHealth(value: number): void;
    }

    export function IsICombatBehaviour(object: any): boolean {
        if (
            object["IsValidTarget"]
            && object["AttackUnit"]
            && object["DefendAndCounter"]
            && object["Defend"]
            && object["DefendAndCounter"]
            && object["GetAttackCommand"]
            && object["CurrentHealth"]
            && object["GetDefenceMultiplier"]
            && object["GetPreview"]
        )
            return true;
        return false;
    }
}

File: WolfDen/Game/Interfaces/ICombatBehaviour.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using WolfDen.Game;

namespace WolfDen.Game.Interfaces
{
    public interface ICombatBehaviour
    {
        UnitType UnitType { get; }
        void AttackUnit(Unit target);
        void DefendAndCounter(ICombatBehaviour source, float attackMultiplier);
        void Defend(ICombatBehaviour source, float attackMultiplier);
    }
}
